### DISCUSSION TITLE
Field that references a record in another system (like Salesforce)

### DISCUSSION THREAD

**Discussion by febbraro** [payloadcms/payload#850] [Q&A]
Thanks for Payload.  In my data model I have a User (and some other records like Company, Project, etc) and I would like to be able to have them reference the canonical data for our organization that lives in Salesforce.  I'm writing to ask what you think might be a best practice for implementing this kind of field?  I would like to be able to search that system for a record of a certain type and then store a reference to its id (and possibly some field data as a local cache).

**Comment by falko100** [comment]
I'd suggest just adding a field to your model. 

This would go into your Users collection file;
```ts
const Users: CollectionConfig = {
  ...,
  fields: [
    // Email added by default
    {
      name: 'name',
      type: 'text',
    },
    {
      label: 'SalesForce ID',
      name: 'crmReference',
      type: 'text',
      // Remember to set the index property to true if you want to search by this field a lot as it will improve query speed.
      index: true,
    },
  ],
}
```

**Reply by jmikrut** [reply]
Nailed it. You can take it a step further and create a custom React component, which integrates with Salesforce and allows the user to select from a list of references. Or, you could write a hook to do this automatically _**for**_ the user based on their email or similar.
